airbyte-integrations/connectors/source-hubspot/source_hubspot/api.py

```diff
@@ -460,6 +465,47 @@ class ContactListStream(Stream):
     limit_field = "count"
 
 
+class DealStageHistoryStream(Stream):
```

Comment:
@jacob-a6 I like how you use stream abstraction here, just few thing to make it even better (and fix some bugs):
* implement `_transform(self, records: Iterable)` which returns only dealstages and set dealId on top level stage record. Note: pass return value too super method and return its result
* set updated_at = 'timestamp'
* remove parse_cursor method as everything will work as usual now
```python
    def _transform(self, records: Iterable) -> Iterable:
        """Preprocess record before emitting"""
        for record in super()._transform(records):
            yield {"dealId": record.get("dealId"), **record.get("properties", {}).get("dealstage")}
```

Reply:
By doing so we can simplify logic and make schema of records emitted from `DealStageHistoryStream` looks more like deal stage records. WDYT?

Reply:
Perfect! Thank you for highlighting `_transform` and the fixes